lambda_src/drivers/process_https.py

```diff
@@ -44,14 +44,17 @@ def process_row(
             raise RuntimeError('url must start with https://')
     else:
         req_host = base_url
-        req_path = url or '/'
 
     req_kwargs = parse_header_dict(kwargs)
-
-    req_headers = {
-        k: v.format(**req_kwargs) for k, v in parse_header_dict(headers).items()
-    }
-    req_headers.setdefault('User-Agent', 'Snowflake Generic External Function 1.0')
+    req_headers = (
+        loads(headers)
+        if headers.startswith('{')
+        else {k: v.format(**req_kwargs) for k, v in parse_header_dict(headers).items()}
```

Comment:
I think we want `v.format(**req_kwargs)` to apply to both `parse_header_dict` and `loads` results